test/lib/Benchmark.js

```diff
@@ -0,0 +1,181 @@
+/* globals XPCOMUtils, Task */
+"use strict";
+
+const {Cu} = require("chrome");
+Cu.import("resource://gre/modules/XPCOMUtils.jsm");
+
+const Microseconds = require("addon/vendor.bundle").Microseconds;
+
+XPCOMUtils.defineLazyModuleGetter(this, "Task",
+                                  "resource://gre/modules/Task.jsm");
+
+/*
+ * A simple benchmark framework that runs on top of jpm test
+ *
+ * Note that the timing resolution for the benchmark is microsecond
+ *
+ * Example:
+
+  function fib(n) {
+    if (n < 2) {
+      return n;
+    }
+
+    return fib(n - 1) + fib(n - 2);
+  }
+
+  exports.test_benchmark = function*() {
+    let bench = new Benchmark(b => {
+      for (let i = 0; i < b.N; i++) {
+        fib(20);
+      }
+    });
+    let result = yield bench.launch();
+    console.log(result);
+  };
+*/
+
+/*
+ * {Params} func is the benchmark function/generator that is of the following form
+ *
+   function(b) {
+     for (let i = 0; i < b.N; i++) {
+       // do your stuff here
+     }
+   }
+ * Note that the framework will decide the best iteration number b.N based on
+ * how fast the target benchmark function runs
+ *
+ * {Params} benchTime is the benchmark time in microseconds, by default it's one second
+ */
+function Benchmark(func, benchTime = 1e6) {
+  this._failed = false;
+  this._benchFunc = func;
+  this._benchTime = benchTime;
+  this._timerOn = false;
+  this._start = 0;
+  this._duration = 0;
+  // N is public to the benchmark functions
+  this.N = 0;
+}
+
+Benchmark.prototype = {
```

Comment:
@piatra I have asked the similar question before. Not sure if there is any difference between those two.

As @dmose mentioned above, I stuck to this one because it's ubiquitous in our codebase.